### PR TITLE
Printing sinks should take ContextShift + EC

### PR DESCRIPTION
**Pull request by monadplus**
Add stdoutLines and stdinLines as commented in the issue https://github.com/functional-streams-for-scala/fs2/issues/1328

### CODE REVIEW

File: io/src/main/scala/fs2/io/io.scala [hunk 1]
```diff
@@ -131,6 +135,25 @@ package object io {
...
+    * blocking so the execution context should be configured appropriately.
+    */
+  def stdoutLines[F[_], O](blockingExecutionContext: ExecutionContext, charset: Charset = utf8Charset)(
+      implicit F: Sync[F],
+      cs: ContextShift[F],
+      show: Show[O]): Pipe[F, O, Unit] =
+    _.map(_.show).through(text.encode(charset)).through(stdout(blockingExecutionContext))
+
+  /** Stream of `String` read asynchronously from standard input decoded in UTF-8. */
+  def stdinUtf8[F[_]](bufSize: Int, blockingExecutionContext: ExecutionContext)(
```
Comment (mpilquist) [inline io/src/main/scala/fs2/io/io.scala]: How about `stdinText` and taking a charset param defaulted to UTF8?

Comment (monadplus) [reply io/src/main/scala/fs2/io/io.scala]: Is this generalization possible ?
I though converting from bytes to string has to be done ad hoc.

Comment (mpilquist) [reply io/src/main/scala/fs2/io/io.scala]: Oh, right, we don't have a `decode(c: CharSet)` method at this point in time.

Comment (monadplus) [reply io/src/main/scala/fs2/io/io.scala]: So how do I proceed ?

**Pull request merged**